python/cudf/cudf/core/index.py

```diff
@@ -1443,7 +1443,21 @@ def __repr__(self):
                     output[:break_idx].replace("'", "") + output[break_idx:]
                 )
             else:
-                output = repr(preprocess.to_pandas())
+                # Too many non-unique categories will cause
+                # the output to take too long. In this case, we
+                # split the categories into data and categories
+                # and generate the repr separately and
+                # merge them.
+                pd_cats = pd.Categorical(
```

Comment:
Do you think it's just worth moving this logic to `CategoricalColumn.to_pandas` and making that conversion faster in general?

Reply:
I feel this could be very brittle and would not want to creep into actual `to_pandas` code. But if someone complains about slowness in `to_pandas` we could always move this logic to `to_pandas`.